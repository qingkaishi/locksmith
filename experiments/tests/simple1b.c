#include <pthread.h>

pthread_mutex_t lock;
int shared;

void* run(void*);

int main() {
  pthread_t t1, t2;

  pthread_mutex_init(&lock, NULL);
  pthread_create(&t1, NULL, run, NULL);
  pthread_create(&t2, NULL, run, NULL);
  return 1;
}

void *run(void *arg) {
  pthread_mutex_lock(&lock);
  shared++;
  pthread_mutex_unlock(&lock);
  return NULL;
}
