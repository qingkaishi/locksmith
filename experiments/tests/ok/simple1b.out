************************* STARTING *************************

shared: &shared:simple1b.c:4
&shared:simple1b.c:4 is protected by:
  concrete lock:simple1b.c:11
  lock:simple1b.c:3
*************************** DONE ***************************
